src/utils/handleSubmit.js

```diff
@@ -0,0 +1,43 @@
+import Popup from "../components/Popup.js";
+
+export function handleSubmit(
+  submitAction,
+  popupInstance, // This should be an instance of Popup or PopupDeleteCard
+  submitText = "Save",
+  loadingText = "Saving..."
+) {
+  // Check if popupInstance is an instance of Popup
+  if (!(popupInstance instanceof Popup)) {
+    console.error("The provided popupInstance is not an instance of Popup.");
+    return Promise.reject("Invalid popup instance");
+  }
+
+  const popup = popupInstance._popupElement;
```

Comment:
* Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.